chainer/dataset/dataset_mixin.py

```diff
@@ -15,18 +19,18 @@ class DatasetMixin(object):
     def __getitem__(self, index):
         """Returns an example or a sequence of examples.
 
-        It implements the standard Python indexing. It uses the
-        :meth:`get_example` method by default, but it may be overridden by the
-        implementation to, for example, improve the slicing performance.
+        It implements the standard Python indexing and numpy like advanced
+        indexing. It uses the :meth:`get_example` method by default, but it may
+        be overridden by the implementation to, for example, improve the
+        slicing performance.
```

Comment:
Since it doesn't support the full features of advanced indexing, I think it is better to clarify which kind of indexing is supported in detail. For example, N-dimensional indexing like `a[0,1]` and boolean array indexing like `a[[True,False]]` are not supported by it.